src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDataColumn.java

```diff
@@ -45,4 +45,27 @@ public String getColumnName() {
     public int getColumnType() {
         return javaSqlType;
     }
+
+    @Override
+    public int hashCode() {
+        int hash = 7;
+        hash = 31 * hash + javaSqlType;
+        hash = 31 * hash + precision;
+        hash = 31 * hash + scale;
+        hash = 31 * hash + numberOfDigitsIntegerPart;
+        hash = 31 * hash + (null != columnName ? columnName.hashCode() : 0);
+        return hash;
+    }
+
+    public boolean equals(Object object) {
+        if (this == object) {
+            return true;
+        }
+
+        if (null != object && object instanceof SQLServerDataColumn) {
+            if (hashCode() == ((SQLServerDataColumn) object).hashCode())

```

Comment:
For both of these equal methods, even if the hashCodes are equal, the actual objects may be different. After validating that hashCodes are equal, we should individually check that the fields are identical between the two objects to return true for equals.